apptools/undo/traitsui/trait_set_command.py

```diff
@@ -0,0 +1,82 @@
+#
+# (C) Copyright 2015 Enthought, Inc., Austin, TX
+# All right reserved.
+#
+# This file is open source software distributed according to the terms in
+# LICENSE.txt
+#
+"""
+TraitSetCommand
+===============
+
+This module provides a simple command which sets a trait value on ``do`` and
+``redo``, resets it to the previous value on ``undo``, and merges multiple
+commands on the same attribute on the same object.
+
+"""
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+# Enthought library imports
+from traits.api import Any, Str
+
+# Local library imports
+from apptools.undo.api import AbstractCommand
+
+
+class TraitSetCommand(AbstractCommand):
+    """ Command subclass which simply sets a trait on an object
+
+    This is somewhat simplistic: it assumes that nothing else is going to
+    hold on to and mutate the old value unexpectedly, which may break down
+    with things like NumPy arrays.  The main purposes is to generate commands
+    from the UndoHandler, so we can be reasonably comfortable that the values
+    are being handled sanely.
+
+    """
+
+    #-------------------------------------------------------------------------
+    # 'TraitSetCommand' interface
+    #-------------------------------------------------------------------------
+
+    #: The name of the data trait we are acting on.
+    trait_name = Str
+
+    #: The new value that the trait is being set to
+    value = Any
+
+    #: The old value of the trait
+    _old_value = Any
+
+    #-------------------------------------------------------------------------
+    # 'ICommand' interface
+    #-------------------------------------------------------------------------
+
+    #: The user-visible name of the command
+    name = Str
+
+    def do(self):
+        # we assume that copy or deep copy isn't required
+        self._old_value = getattr(self.data, self.trait_name)
```

Comment:
This will fail ungracefully for `Event` traits (`Button` presses). While it isn't possible to automatically handle the undoability of such UI actions, we need to avoid making a `TraitSetCommand` in such cases.